### DISCUSSION TITLE
Background Position vs Background Size

### DISCUSSION THREAD

**Discussion by danspratling** [tailwindlabs/tailwindcss#11807] [General]
Hey. I've just run into a (small) problem with the background position vs background size properties.

I'll phrase this as a question.

What would you expect `bg-[200%]` to do?

Based on the docs, it looks like this could be one of two answers

```
// either
background-size: 200%;
//or
background-position: 200%;
```

The actual answer is that background position seems to be picked first if a % is used, while arguably size is more likely for a % over 100. Because both of these are technically valid there is potential for confusion.

So if you want to do effectively `bg-size-[200%]` there doesn't appear to be a means to allow this, despite _technically_ supporting the option.

If there _is_ a supported method for implementing both bg size and position as a percent at the same time, I'd love to see an example added to both the page and the arbitrary values docs.

**Comment by wongjn** [comment]
With regards to ambigious nature of the `bg-` prefix, you could consider "typing" the arbitrary value[^1][^2].

For `background-position`, set the typing as `position`:

https://github.com/tailwindlabs/tailwindcss/blob/4c34b2ed34a162750df581af460121322132e88d/src/corePlugins.js#L1905-L1907

Also of note that `preferOnConflict` is set to `true`, so `background-position` will be chosen for an ambigious value that could be for `background-position` or `background-size`.

Or for `background-size`, set the typing as `length`, `percentage` or `size`:

https://github.com/tailwindlabs/tailwindcss/blob/4c34b2ed34a162750df581af460121322132e88d/src/corePlugins.js#L1884-L1886

If you would like to contribute to the documentation, feel free to send a pull request to the [documentation website repository](https://github.com/tailwindlabs/tailwindcss.com).

[^1]: https://tailwindcss.com/docs/background-size#arbitrary-values
[^2]: https://tailwindcss.com/docs/adding-custom-styles#resolving-ambiguities

**Comment by leetcow** [comment]
just ran into this today after trying to set background size with `bg-[100%_2px]` and the position being set instead
